test/distributed/fsdp/test_fsdp_grad_acc.py

```diff
@@ -132,8 +162,13 @@ def permute_tensor(x: torch.Tensor):
 
     @skip_if_lt_x_gpu(2)
     @parametrize(
-        "num_iters_to_acc",
-        [2, 4],
+        "configs",
+        [
+            [_GradAccConfig(True, 4)],
```

Comment:
to reduce # of tests, can we just have:

use_context = true, interval={2,4}
use_context = false, interval={2,4}

Reply:
Given your comments last time, I thought about how to keep the set of tests minimal. I feel like these 4 configs test distinct things:
1) Gradient accumulation with `no_sync()`
2) Gradient accumulation without `no_sync()`
3) Gradient accumulation interleaving `no_sync()` and without `no_sync()`, where the last iteration before synchronizing gradients is in `no_sync()`
4) Gradient accumulation interleaving `no_sync()` and without `no_sync()`, where the last iteration before synchronizing gradients is outside `no_sync()`

My concern is that I do not want to overfit the current working implementation. It not obvious to me that 1) and 2) imply 3) and 4), and if we only had 3) and 4) and they break, then we would probably end up rewriting 1) and 2).

Reply:
Sounds good! Considering you've gone through the tradeoff this makes sense to leave as is